src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -52,46 +53,64 @@ public partial class NodeViewModel : ViewModelBase
 
         #region public members
 
+        /// <summary>
+        /// Returns NodeModel ID
+        /// </summary>
+        public Guid Id
```

Comment:
@QilongTang lets talk about this - I'm having trouble reasoning about how this will interact with the GUID property on the nodeModel - specifically the converter I applied there to the GUID to ID.

```
        /// <summary>
        /// Id for this node, must be unique within the graph.
        /// </summary>
        [JsonProperty("Id")]
        [JsonConverter(typeof(IdToGuidConverter))]
        public override Guid GUID
        {
            get
            {
                return base.GUID;
            }
            set
            {
                base.GUID = value;
            }
        }
```